_nix

```diff
@@ -1392,6 +1392,24 @@ NIX_FILE
     done
     return 1
 }
+
+# Configure 'nix' completion provider: nix>=2.4 contains its own completions,
+# prefer those.
+function _configure_modern_nix_cli_completion () {
+    local version_line versions maj_ver min_ver
+    version_line="$(nix --version || true)"
+    IFS='.' read -r -a versions <<< "${version_line##* }"
+    maj_ver="${versions[0]}"
+    min_ver="${versions[1]}"
+    if [[ "$maj_ver" -ge 2 ]] && [[ "$min_ver" -ge 4 ]]; then
+        # Nix has completion
+        source "$(dirname "$(realpath "$(command -v nix)")")"/../share/bash-completion/completions/nix
+        return
+    fi
+    # Fallback for old Nix
+    complete -F _nix_completion nix
```

Comment:
```suggestion
    if (( maj_ver < 2 || (maj_ver == 2 && min_ver < 4) )); then
        # Older versions of Nix didn't have completion for the nix command
        complete -F _nix_completion nix
    fi
```

Fixed your version comparison (it would have treated 3.0 as old).

Also, it's not our responsibility to source Nix's completion files (we can't assume they're in that location either).

Reply:
> Fixed your version comparison (it would have treated 3.0 as old).

Thanks!